tests/ui/mixup.rs

```diff
@@ -0,0 +1,14 @@
+extern crate compact_arena;
+
+use compact_arena::in_nano_arena;
+
+fn main() {
+    in_nano_arena!(a, {
+        in_nano_arena!(b, {
+            let _ = a.add(1usize);
```

Comment:
Why did you add two entities to `a`?

Reply:
To really get an out of bounds access.

Reply:
But you don't need an out of bounds access, you need compile time error that `x` cannot be used as an index of `b`